spyder/plugins/plots/widgets/tests/test_plots_widgets.py

```diff
@@ -0,0 +1,197 @@
+    assert osp.exists(spy_figname)
+
+    # Compare the figure created with matplotlib with the one created with our
+    # figure browser.
+    with open(spy_figname, "rb") as figfile:
+        spyfig = figfile.read()
+    assert mplfig == spyfig
+
+
+def test_clear_current_figure(figbrowser, tmpdir):
+    """
+    Test that clearing the current figure works as expected.
+    """
+    # Open some figures in the figure browser.
+    figs = add_figures_to_browser(figbrowser, 2, tmpdir)
+
+    # Remove the first figure.
+    figbrowser.close_figure()
+    assert len(figbrowser.thumbnails_sb._thumbnails) == 1
+    assert figbrowser.thumbnails_sb.get_current_index() == 0
+    assert figbrowser.thumbnails_sb.current_thumbnail.canvas.fig == figs[0]
+    assert figbrowser.figviewer.figcanvas.fig == figs[0]
+
+    # Remove the last figure.
+    figbrowser.close_figure()
+    assert len(figbrowser.thumbnails_sb._thumbnails) == 0
+    assert figbrowser.thumbnails_sb.get_current_index() == -1
+    assert figbrowser.thumbnails_sb.current_thumbnail is None
+    assert figbrowser.figviewer.figcanvas.fig is None
+
+
+def test_clear_all_figures(figbrowser, tmpdir):
+    """
+    Test that clearing all figures displayed in the thumbnails scrollbar
+    works as expected.
+    """
+    # Open some figures in the figure browser.
+    add_figures_to_browser(figbrowser, 3, tmpdir)
+
+    # Close all previously opened figures.
+    figbrowser.close_all_figures()
+    assert len(figbrowser.thumbnails_sb._thumbnails) == 0
+    assert figbrowser.thumbnails_sb.get_current_index() == -1
+    assert figbrowser.thumbnails_sb.current_thumbnail is None
+    assert figbrowser.figviewer.figcanvas.fig is None
+
+
+def test_go_prev_next_figure(figbrowser, tmpdir):
+    """
+    Test that the go to previous and next figure actions are working
+    as expected.
+    """
+    # Open some figures in the figure browser.
+    figs = add_figures_to_browser(figbrowser, 3, tmpdir)
+    assert figbrowser.thumbnails_sb.get_current_index() == 2
+    assert figbrowser.thumbnails_sb.current_thumbnail.canvas.fig == figs[2]
+    assert figbrowser.figviewer.figcanvas.fig == figs[2]
+
+    figbrowser.go_next_thumbnail()
+    assert figbrowser.thumbnails_sb.get_current_index() == 0
+    assert figbrowser.thumbnails_sb.current_thumbnail.canvas.fig == figs[0]
+    assert figbrowser.figviewer.figcanvas.fig == figs[0]
+
+    figbrowser.go_previous_thumbnail()
+    assert figbrowser.thumbnails_sb.get_current_index() == 2
+    assert figbrowser.thumbnails_sb.current_thumbnail.canvas.fig == figs[2]
+    assert figbrowser.figviewer.figcanvas.fig == figs[2]
+
+    figbrowser.go_previous_thumbnail()
+    assert figbrowser.thumbnails_sb.get_current_index() == 1
+    assert figbrowser.thumbnails_sb.current_thumbnail.canvas.fig == figs[1]
+    assert figbrowser.figviewer.figcanvas.fig == figs[1]
+
+
+if __name__ == "__main__":
+    import os
+    pytest.main([os.path.basename(__file__), '-vv', '-rw'])
```

Comment:
So this is our standard ``if __name_ == "__main__":`` block now? Just want to make sure I'm conforming to current conventions. Thanks; really happy to see these tests!

Reply:
No this is not, I'll change that when I'm finished. It's just more convenient for me to test my test while I'm writing them.